### PR TITLE
Add strncmp() conformance tests and assertion.xml

### PR DESCRIPTION
**Pull request by padmashree9110**
Adds
1> 1-1, 2-1, 3-1 tests
2> assertion.xml
3> coverage.txt

Change-Id: I41881d59fb91e543651fe7ef87984b55ef1692cc

### CODE REVIEW

**Comment by anitha-suresh** [comment]
@metan-ucw : Could you please review/merge the patch.

**Comment by padmashree9110** [comment]
@ngie-eign Can you please review this.

**Comment by padmashree9110** [comment]
@ngie-eign Can you please review this.

**Comment by padmashree9110** [comment]
@ngie-eign Please review the updated patch.

**Comment by padmashree9110** [comment]
@ngie-eign Can u please review the patch.

**Comment by padmashree9110** [comment]
@ngie-eign Please review/merge the update patch.

File: testcases/open_posix_testsuite/conformance/interfaces/strncmp/1-1.c [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+#include <stdlib.h>
+#include <unistd.h>
+#include <stdbool.h>
+#include <ctype.h>
+#include "posixtest.h"
+
+#define STRING_MAX_LEN 50000
+#define STEP_COUNT 2000
+#define TRUE 1
+#define FALSE 0
```
Comment (metan-ucw) [inline testcases/open_posix_testsuite/conformance/interfaces/strncmp/1-1.c]: Here as well.

File: testcases/open_posix_testsuite/conformance/interfaces/strncmp/1-1.c [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+		} else {
+			//Generates a lowercase character
+			if (!islower(output_string[i]))
+				i--;
+		}
+	}
+	output_string[len] = '\0';
+	return output_string;
+}
+
```
Comment (metan-ucw) [inline testcases/open_posix_testsuite/conformance/interfaces/strncmp/1-1.c]: Here as well, should be put into an header.

File: testcases/open_posix_testsuite/conformance/interfaces/strncmp/1-1.c [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+		num_bytes = rand() % i + 1;
+		sample_str_2 = random_string(i, true);
+		sample_str_1 = random_string(i, false);
+
+		ret = strncmp(sample_str_1, sample_str_2, num_bytes);
+
+		if (ret <= 0) {
+			printf(TNAME " Test Failed, return value is not as expected, "
+						"for string1 greater than string2. Expected > 0,"
+						"but got %d.\n", ret);
```
Comment (metan-ucw) [inline testcases/open_posix_testsuite/conformance/interfaces/strncmp/1-1.c]: Here as well, message could be shorter.

File: testcases/open_posix_testsuite/conformance/interfaces/strncmp/1-1.c [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+			printf("str1=%s\n str2=%s\n", sample_str_1, sample_str_2);
+			exit(PTS_FAIL);
+		}
+		free(sample_str_1);
+		free(sample_str_2);
+	}
+	printf(TNAME " Test Passed, return value is as expected, "
+				"for string1 greater than string2.\n");
```
Comment (metan-ucw) [inline testcases/open_posix_testsuite/conformance/interfaces/strncmp/1-1.c]: Here as well.

File: testcases/open_posix_testsuite/conformance/interfaces/strncmp/2-1.c [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+		sample_str_1[i] = output_char;
+		if (i <= num_bytes)
+			sample_str_2[i] = output_char;
+		else
+			sample_str_2[i] = output_char + 1;
+	}
+	sample_str_1[i] = '\0';
+	sample_str_2[i] = '\0';
+}
+
```
Comment (metan-ucw) [inline testcases/open_posix_testsuite/conformance/interfaces/strncmp/2-1.c]: This function is shared with the strcmp() test, so it should be put into an common header as well.

File: testcases/open_posix_testsuite/conformance/interfaces/strncmp/2-1.c [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+			exit(PTS_UNRESOLVED);
+		}
+		num_bytes = rand() % i + 1;
+		random_string(sample_str_1, sample_str_2, i, num_bytes);
+
+		ret = strncmp(sample_str_1, sample_str_2, num_bytes);
+		if (ret != 0) {
+			printf(TNAME " Test Failed, return value is not as expected, "
+					"for string1 equal to string2. Expected 0,"
+					"but got %d.\n", ret);
```
Comment (metan-ucw) [inline testcases/open_posix_testsuite/conformance/interfaces/strncmp/2-1.c]: Here as well.

File: testcases/open_posix_testsuite/conformance/interfaces/strncmp/2-1.c [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+			exit(PTS_FAIL);
+		}
+		free(sample_str_1);
+		free(sample_str_2);
+	}
+	printf(TNAME " Test Passed, return value is as expected, "
+			"for string1 equal to string2.\n");
```
Comment (metan-ucw) [inline testcases/open_posix_testsuite/conformance/interfaces/strncmp/2-1.c]: Here as well.

File: testcases/open_posix_testsuite/conformance/interfaces/strncmp/3-1.c [hunk 8]
```diff
@@ -0,0 +1,90 @@
...
+#include <stdlib.h>
+#include <unistd.h>
+#include <stdbool.h>
+#include <ctype.h>
+#include "posixtest.h"
+
+#define STRING_MAX_LEN 50000
+#define STEP_COUNT 2000
+#define TRUE 1
+#define FALSE 0
```
Comment (metan-ucw) [inline testcases/open_posix_testsuite/conformance/interfaces/strncmp/3-1.c]: Here as well.

File: testcases/open_posix_testsuite/conformance/interfaces/strncmp/3-1.c [hunk 9]
```diff
@@ -0,0 +1,90 @@
...
+				i--;
+		} else {
+			//Generates a lowercase character
+			if (!islower(output_string[i]))
+				i--;
+		}
+	}
+	output_string[len] = '\0';
+	return output_string;
+}
```
Comment (metan-ucw) [inline testcases/open_posix_testsuite/conformance/interfaces/strncmp/3-1.c]: Here as well.

File: testcases/open_posix_testsuite/conformance/interfaces/strncmp/3-1.c [hunk 10]
```diff
@@ -0,0 +1,90 @@
...
+		num_bytes = rand() % i + 1;
+		sample_str_2 = random_string(i, false);
+		sample_str_1 = random_string(i, true);
+
+		ret = strncmp(sample_str_1, sample_str_2, num_bytes);
+
+		if (ret >= 0) {
+			printf(TNAME " Test Failed, return value is not as expected, "
+						"for string1 less than string2. Expected < 0,"
+						"but got %d.\n", ret);
```
Comment (metan-ucw) [inline testcases/open_posix_testsuite/conformance/interfaces/strncmp/3-1.c]: Here as well.

File: testcases/open_posix_testsuite/conformance/interfaces/strncmp/3-1.c [hunk 11]
```diff
@@ -0,0 +1,90 @@
...
+			exit(PTS_FAIL);
+		}
+		free(sample_str_1);
+		free(sample_str_2);
+	}
+	printf(TNAME " Test Passed, return value is as expected, "
+				"for string1 less than string2.\n");
```
Comment (metan-ucw) [inline testcases/open_posix_testsuite/conformance/interfaces/strncmp/3-1.c]: Here as well.

**Comment by metan-ucw** [comment]
This is the same as strcmp() tests, generally looks good, needs a few minor things to be sorted out.

**Pull request closed without merging**